services/core-api/app/api/mines/reports/models/mine_report.py

```diff
@@ -1,43 +1,97 @@
+    mine_report_guid = db.Column(UUID(as_uuid=True), server_default=FetchedValue(), nullable=False)
+
     mine_report_definition_id = db.Column(
-        db.Integer, db.ForeignKey('mine_report_definition.mine_report_definition_id'))
+        db.Integer,
+        db.ForeignKey('mine_report_definition.mine_report_definition_id'),
+        nullable=False)
     mine_report_definition = db.relationship('MineReportDefinition', lazy='joined')
-    mine_guid = db.Column(UUID(as_uuid=True), db.ForeignKey('mine.mine_guid'))
+    mine_report_definition_guid = association_proxy('mine_report_definition',
+                                                    'mine_report_definition_guid')
+    report_name = association_proxy('mine_report_definition', 'report_name')
+    mine_report_category_xref = db.relationship(
+        'MineReportCategoryXref',
+        lazy='joined',
+        primaryjoin=
+        'MineReport.mine_report_definition_id==MineReportCategoryXref.mine_report_definition_id',
+        foreign_keys=mine_report_definition_id,
+        uselist=True,
+        order_by='MineReportCategoryXref.mine_report_category')
+    mine_report_category = association_proxy('mine_report_category_xref', 'mine_report_category')
+
+    mine_guid = db.Column(UUID(as_uuid=True), db.ForeignKey('mine.mine_guid'), nullable=False)
+    mine_table = db.relationship('Mine', lazy='joined')
+    mine_name = association_proxy('mine_table', 'mine_name')
+    mine_region = association_proxy('mine_table', 'mine_region')
+    major_mine_ind = association_proxy('mine_table', 'major_mine_ind')
+
     permit_id = db.Column(db.Integer, db.ForeignKey('permit.permit_id'))
     permit = db.relationship('Permit', lazy='selectin')
+    permit_guid = association_proxy('permit', 'permit_guid')
+
     received_date = db.Column(db.DateTime)
-    due_date = db.Column(db.DateTime, nullable=False)
+    due_date = db.Column(db.DateTime)
     submission_year = db.Column(db.Integer)
+
     deleted_ind = db.Column(db.Boolean, server_default=FetchedValue(), nullable=False)
 
     mine_report_submissions = db.relationship(
         'MineReportSubmission',
         lazy='joined',
         order_by='asc(MineReportSubmission.mine_report_submission_id)',
-    )
+        uselist=True)
 
     created_by_idir = db.Column(db.String, nullable=False, default=User().get_user_username)
 
-    mine_report_definition_guid = association_proxy('mine_report_definition',
-                                                    'mine_report_definition_guid')
-    report_name = association_proxy('mine_report_definition', 'report_name')
-    report_name = association_proxy('mine_report_definition', 'report_name')
-    permit_guid = association_proxy('permit', 'permit_guid')
+    @hybrid_property
+    def mine_report_submission_status_code(self):
+        if self.mine_report_submissions and len(self.mine_report_submissions) > 0:
+            return self.mine_report_submissions[len(self.mine_report_submissions) -
```

Comment:
use array slicing to access last element of the array, `self.mine_report_submissions[-1].mine_report_submission_status_code` i'm wondering if there is a way to define a relationship so that it's eagerly loaded.

Also, you are using the status of the submission with the highest id to represent the status of the report. can you rename the attribute to match -> `mine_report_status_code`

Reply:
It doesn't really need to be eagerly loaded since this is only used in filtering/sorting?

Reply:
It doesn't really need to be eagerly loaded since its only being used for sorting/filtering?